src/commonMain/kotlin/teksturepako/pakku/api/actions/update/Update.kt

```diff
@@ -34,29 +35,36 @@ suspend fun updateMultipleProjectsWithFiles(
             }
     }
 
-    return@coroutineScope platforms.fold(projects.map { it.copy(files = mutableSetOf()) }.toMutableSet()) { acc, platform ->
+    val projectsToFiles = projects.associateTo(mutableMapOf()) { it.copy(files = mutableSetOf()) to it.files }
+
+    return@coroutineScope platforms.fold(projectsToFiles.keys.toMutableSet()) { acc, platform ->
 
         val listOfIds = projects.mapNotNull { it.id[platform.serialName] }
 
-        platform.requestMultipleProjectsWithFiles(mcVersions, loaders, listOfIds, numberOfFiles)
-            .inheritPropertiesFrom(configFile)
-            .forEach { newProject ->
+        platform.requestMultipleProjectsWithFiles(mcVersions, loaders, listOfIds, Int.MAX_VALUE)
+            .inheritPropertiesFrom(configFile).forEach { newProject ->
                 acc.find { accProject ->
                     accProject.slug[platform.serialName] == newProject.slug[platform.serialName]
-                }?.let { accProject ->
+                }?.also { accProject ->
                     // Combine projects
-                    (accProject + newProject).get()?.let x@ { combinedProject ->
-                        if (combinedProject.hasNoFiles()) return@x // Do not update project if files are missing
-                        acc -= accProject
-                        acc += combinedProject
-                    }
+                    val accFiles = projectsToFiles[accProject]!!
+                    val accPublished = accFiles.find { it.type == platform.serialName }?.dataPublished
+                    if (accPublished != null && accPublished != Instant.MIN)
```

Comment:
This probably is not backwards compatible. Old lock files don't have the "date published".

Reply:
It will be `Instant.MIN` since there is a default value assigned